src/main/java/com/ibm/watson/modelmesh/processor/AsyncPayloadProcessor.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.ibm.watson.modelmesh.processor;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class AsyncPayloadProcessor implements PayloadProcessor {
+
+    private final PayloadProcessor delegate;
+
+    private final Queue<Payload> payloads = new ConcurrentLinkedQueue<>();
+
+    public AsyncPayloadProcessor(PayloadProcessor delegate) {
+        this(delegate, 1, TimeUnit.MINUTES, Executors.newScheduledThreadPool(1));
+    }
+
+    public AsyncPayloadProcessor(PayloadProcessor delegate, int delay, TimeUnit timeUnit,
+                                 ScheduledExecutorService executorService) {
+        this.delegate = delegate;
+        executorService.scheduleWithFixedDelay(() -> {
```

Comment:
I think the continuous loop within an always running thread would kill CPU usage. In this way we have a dedicated thread running at fixed delays without continuously polling payload queues.

Reply:
now I get what you mean, so the queue implementation should have a _blocking_ `poll` implementation (hence creating a `FixedSizeLinkedBlockingDeque`, for example), because the current one has a non blocking `poll`.

Then the logging of _dropped_ payloads would be taken care of by a separate scheduled thread.